File: app/src/main/java/com/example/gallery/Full_Image_Activity.java
package com.example.gallery;

import static android.Manifest.permission.READ_EXTERNAL_STORAGE;
import static android.Manifest.permission.WRITE_EXTERNAL_STORAGE;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.net.Uri;
import android.os.Bundle;
import android.os.StrictMode;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

public class Full_Image_Activity extends AppCompatActivity {


    int position;
    ImageView imageView;
    String imageLink;
    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_full_image);

        ActivityCompat.requestPermissions(this, new String[]{READ_EXTERNAL_STORAGE, WRITE_EXTERNAL_STORAGE}, PackageManager.PERMISSION_GRANTED);

        StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
        StrictMode.setVmPolicy(builder.build());

        imageView = findViewById(R.id.fullimageViewid);
        button = findViewById(R.id.button);

        Bundle bundle = getIntent().getExtras();
        if (bundle != null){
            position = bundle.getInt("pos",0);
           imageLink =  bundle.getString("image");
        }

        imageView.setImageURI(Uri.parse(imageLink));

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                Intent intent = new Intent(Intent.ACTION_SEND);
//                // putting uri of image to be shared
                intent.putExtra(Intent.EXTRA_STREAM, Uri.parse(imageLink));
//                // adding text to share
                intent.putExtra(Intent.EXTRA_TEXT, "Add Caption");
//                // Add subject Here
                intent.putExtra(Intent.EXTRA_SUBJECT, "Add Subject");
//                // setting type to image
                intent.setType("image/png");
//                // calling startActivity() to share
                startActivity(Intent.createChooser(intent, "Share Via"));
            }
        });
    }
}

File: app/src/main/java/com/example/gallery/FullImageActivity.kt
package com.example.gallery

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.core.app.ActivityCompat
import android.Manifest.permission
import android.content.pm.PackageManager
import android.os.StrictMode.VmPolicy
import android.os.StrictMode
import android.content.Intent
import android.net.Uri
import androidx.databinding.DataBindingUtil
import com.example.gallery.databinding.ActivityFullImageBinding

class FullImageActivity : AppCompatActivity() {
    private lateinit var binding: ActivityFullImageBinding
    private var position = 0
    private lateinit var imageLink: String
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_full_image)
        ActivityCompat.requestPermissions(
            this,
            arrayOf(permission.READ_EXTERNAL_STORAGE, permission.WRITE_EXTERNAL_STORAGE),
            PackageManager.PERMISSION_GRANTED
        )
        val builder = VmPolicy.Builder()
        StrictMode.setVmPolicy(builder.build())
        val bundle = intent.extras
        if (bundle != null) {
            position = bundle.getInt("pos", 0)
            imageLink = bundle.getString("image").toString()
        }
        binding.fullimageViewid.setImageURI(Uri.parse(imageLink))
        binding.button.setOnClickListener{
            val intent = Intent(Intent.ACTION_SEND)
            //                // putting uri of image to be shared
            intent.putExtra(Intent.EXTRA_STREAM, Uri.parse(imageLink))
            //                // adding text to share
            //                // Add subject Here
            intent.putExtra(Intent.EXTRA_SUBJECT, "Add Subject")
            //                // setting type to image
            intent.type = "image/*"
            //                // calling startActivity() to share
            startActivity(Intent.createChooser(intent, "Share Via"))
        }
    }
}